src/functions.sh

```diff
@@ -321,6 +326,54 @@ get_cake_lla_string() {
 }
 
 
+# centralize the implementation for the default sqm_start sqeuence
+# the individual sqm_start function only need to do the individually
+# necessary checking.
+# This expects the calling script to supply both an egress() and ingress() function
+# and will warn if they are missing
+sqm_start_default() {
+    #sqm_error "sqm_start_default"
+    [ -n "$IFACE" ] || return 1
+    
+    fn_exists sqm_prepare_script
+    if [ "$?" -eq "0" ]; then
```

Comment:
Didn't notice this before, but why are you not just doing

if fn_exists sqm_prepare_script; then

instead of then putting the return code into a [] test?

Reply:
Good question, I have not thought about this. If you have prefences one way or the other, I am happy to change this.

Reply:
Cool, please do :)

Reply:
Okay changed, tested, committed, and pushed... please check!